--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -9,6 +9,7 @@
 # time.
 genrule(
     name = "generate_headers",
+    target_compatible_with = ["@platforms//os:linux"],
     srcs = [
         "Makefile.common",
         "liburing.spec",
@@ -37,6 +38,7 @@ genrule(

 cc_library(
     name = "uring",
+    target_compatible_with = ["@platforms//os:linux"],
     srcs = [
         "config-host.h",
         "src/include/liburing/compat.h",
@@ -70,6 +72,7 @@ cc_library(

 alias(
     name = "liburing",
+    target_compatible_with = ["@platforms//os:linux"],
     actual = ":uring",
     visibility = ["//visibility:public"],
 )
