--- a/configure
+++ b/configure
@@ -470,8 +470,8 @@ else
 fi

 MAKE_PRINT_VARS="include Makefile.common\nprint-%: ; @echo \$(\$*)\n"
-VERSION_MAJOR=$(env echo -e "$MAKE_PRINT_VARS" | make -s --no-print-directory -f - print-VERSION_MAJOR)
-VERSION_MINOR=$(env echo -e "$MAKE_PRINT_VARS" | make -s --no-print-directory -f - print-VERSION_MINOR)
+VERSION_MAJOR=$(printf "%b\n" "$MAKE_PRINT_VARS" | make -s --no-print-directory -f - print-VERSION_MAJOR)
+VERSION_MINOR=$(printf "%b\n" "$MAKE_PRINT_VARS" | make -s --no-print-directory -f - print-VERSION_MINOR)

 if [ "$VERSION_MAJOR" = "" ]; then
     echo "ERROR: VERSION_MAJOR could not be extracted from Makefile" 1>&2
