diff --git a/.bazelrc b/.bazelrc
--- a/.bazelrc
+++ b/.bazelrc
@@ -27,6 +27,8 @@ build --copt=-ffile-compilation-dir=. --host_copt=-ffile-compilation-dir=.
 build:system-clang --extra_toolchains=@local_config_cc_toolchains//:all
 build:system-clang --action_env=BAZEL_COMPILER=clang
 build:system-clang --cxxopt=-std=c++23 --host_cxxopt=-std=c++23
 build:system-clang --linkopt -fuse-ld=lld --host_linkopt -fuse-ld=lld
+build:system-clang --host_linkopt -stdlib=libc++
+build:system-clang --host_linkopt --unwindlib=libgcc
 # use a compiler name that doesn't symlink to ccache
 build:system-clang-19 --config=system-clang
 build:system-clang-19 --action_env=CC=clang-19 --host_action_env=CC=clang-19
