diff --git a/src/v/version/expand_with_stamp_vars.bzl b/src/v/version/expand_with_stamp_vars.bzl
index 110870cc24..579aa97389 100644
--- a/src/v/version/expand_with_stamp_vars.bzl
+++ b/src/v/version/expand_with_stamp_vars.bzl
@@ -8,11 +8,13 @@ is not stamped (ie. dev builds).
 
 def _expand_with_stamp_vars(ctx):
     toolchain = ctx.toolchains[Label("@rules_python//python:toolchain_type")]
+    interpreter = toolchain.py3_runtime.interpreter
+    tools = [ctx.executable._tool] + ([interpreter] if interpreter else [])
     ctx.actions.run(
         outputs = [ctx.outputs.out],
         inputs = [ctx.file.defaults_file, ctx.file.template, ctx.info_file],
-        tools = [ctx.executable._tool, toolchain.py3_runtime.interpreter],
-        executable = ctx.executable._tool.path,
+        tools = tools,
+        executable = ctx.executable._tool,
         arguments = [
             "--template",
             ctx.file.template.path,
@@ -23,7 +25,7 @@ def _expand_with_stamp_vars(ctx):
             ctx.outputs.out.path,
         ],
         mnemonic = "ExpandStampVars",
-        use_default_shell_env = False,
+        use_default_shell_env = True,
     )
     return DefaultInfo(
         files = depset([ctx.outputs.out]),
