diff --git a/src/v/version/expand_with_stamp_vars.bzl b/src/v/version/expand_with_stamp_vars.bzl
--- a/src/v/version/expand_with_stamp_vars.bzl
+++ b/src/v/version/expand_with_stamp_vars.bzl
@@ -23,7 +23,7 @@ def _expand_with_stamp_vars(ctx):
             ctx.outputs.out.path,
         ],
         mnemonic = "ExpandStampVars",
-        use_default_shell_env = False,
+        use_default_shell_env = True,
     )
     return DefaultInfo(
         files = depset([ctx.outputs.out]),
